src/resolve/resolved-dns-rr.c

```diff
@@ -826,6 +827,114 @@ static char *format_txt(DnsTxtItem *first) {
+
+                        r = strv_push(&values_strv, alpn);
+                        if (r < 0)
+                                return NULL;
+
+                        offset += sz;
+                }
+                value = strv_join(values_strv, ",");
+                if (!value)
+                        return NULL;
+                break;
+
+        }
+        case DNS_SVC_PARAM_KEY_PORT: {
+                uint16_t port = unaligned_read_be16(i->value);
+                r = asprintf(&value, "%" PRIu16, port);
+                if (r < 0)
+                        return NULL;
+                return TAKE_PTR(value);
+        }
+        case DNS_SVC_PARAM_KEY_IPV4HINT: {
+                const struct in_addr *addrs = (struct in_addr*) &i->value;
+                _cleanup_strv_free_ char **values_strv = NULL;
+                for (size_t n = 0; n < i->length / sizeof (struct in_addr); n++) {
+                        char *addr;
+                        r = in_addr_to_string(AF_INET, (const union in_addr_union*) &addrs[n], &addr);
+                        if (r < 0)
+                                return NULL;
+                        r = strv_push(&values_strv, addr);
+                        if (r < 0)
+                                return NULL;
+                }
+                return strv_join(values_strv, ",");
+        }
+        case DNS_SVC_PARAM_KEY_IPV6HINT: {
+                const struct in6_addr *addrs = (struct in6_addr*) &i->value;
+                _cleanup_strv_free_ char **values_strv = NULL;
+                for (size_t n = 0; n < i->length / sizeof (struct in6_addr); n++) {
+                        char *addr;
+                        r = in_addr_to_string(AF_INET6, (const union in_addr_union*) &addrs[n], &addr);
+                        if (r < 0)
+                                return NULL;
+                        r = strv_push(&values_strv, addr);
+                        if (r < 0)
+                                return NULL;
+                }
+                return strv_join(values_strv, ",");
+        }
+        default: {
+                value = decescape((char *)&i->value, " ,", i->length);
+                if (!value)
+                        return NULL;
+                break;
+        }
+        }
+
+        char *qvalue;
+        r = asprintf(&qvalue, "\"%s\"", value);
+        if (r < 0)
+                return NULL;
+        return qvalue;
+}
+
+static char *format_svc_param(DnsSvcParam *i) {
+        const char *key = FORMAT_DNS_SVC_PARAM_KEY(i->key);
+
+        _cleanup_free_ char *value = NULL;
+
+        if (i->length == 0)
+                return strdup(key);
+
+        value = format_svc_param_value(i);
+        if (!value)
+                return NULL;
+
+        return strjoin(key, "=", value);
+}
+
+static char *format_svc_params(DnsSvcParam *first) {
+        _cleanup_strv_free_ char **params = NULL;
+
+        LIST_FOREACH(params, i, first) {
+                int r;
```

Comment:
in our codebase `int r` gets this special privilege to be always declared on the outermost scope, since it is the variable we always store error results in. hence, please move this up one scope.